### PR TITLE
Add option to allow IP subnets

### PR DESCRIPTION
**Pull request by stanek-michal**
The allowed_subnets map will contain user-defined IPs and subnets.
The allowed_IPs map will now only be used by the KprobeConnectHook program
to dynamically add destination IPs that allowed processes want to connect to.

The TcFilter network filter checks both maps and allows the packet to
be transmitted if it finds a match in either map.

### CODE REVIEW

File: GPL/HostIsolation/TcFilter/TcFilter.bpf.c [hunk 1]
```diff
@@ -32,6 +32,14 @@
 // you took NULL for granted didn't you :)
 #define NULL 0
 
+#define BPF_F_NO_PREALLOC	(1U << 0)
+
+/* Key of an a BPF_MAP_TYPE_LPM_TRIE entry */
+struct bpf_lpm_trie_key {
+	__u32	prefixlen;	/* up to 32 for AF_INET, 128 for AF_INET6 */
+	__u32	data;	/* Arbitrary size */
```
Comment (nicholasberlin) [inline GPL/HostIsolation/TcFilter/TcFilter.bpf.c:40]: Perhaps this should be large enough to hold an ipv6 address.

File: non-GPL/HostIsolationMapsUtil/UpdateMaps.c [hunk 2]
```diff
@@ -44,23 +66,64 @@ ebpf_map_allowed_pids_add(uint32_t pid)
...
 int
 ebpf_map_allowed_pids_clear()
 {
-    return ebpf_clear_map(EBPF_ALLOWED_PIDS_MAP_PATH);
+    return ebpf_clear_map(EBPF_ALLOWED_PIDS_MAP_PATH, EBPF_MAP_ALLOWED_PIDS);
 }
 
 static int
-ebpf_update_map(const char *map_path, uint32_t key, uint32_t val)
+ebpf_create_map(const char *map_path,
```
Comment (nicholasberlin) [inline non-GPL/HostIsolationMapsUtil/UpdateMaps.c:91]: `map_path` is not used in this function. perhaps remove it?

**Review by nicholasberlin — APPROVED** [review]
Some clean up / simplification suggestions but LGTM

File: non-GPL/HostIsolationMapsUtil/UpdateMaps.c (hunk 2 above)
Comment (stanek-michal) [reply non-GPL/HostIsolationMapsUtil/UpdateMaps.c:91]: I'll remove it

File: non-GPL/HostIsolationMapsUtil/UpdateMaps.c [hunk 3]
```diff
@@ -44,23 +65,63 @@ ebpf_map_allowed_pids_add(uint32_t pid)
...
+                             ebpf_maps[map_id].map_flags);
+
+    if (fd < 0)
+    {
+        ebpf_log("Error creating map\n");
+        rv = -1;
+        goto cleanup;
+    }
+
+    *map_fd = fd;
```
Comment (dwhyrock) [inline non-GPL/HostIsolationMapsUtil/UpdateMaps.c:117]: minor nitpick about the lack of a NULL check on `map_fd`, although it appears this function is only used internally in this file w/ refs to local variables.  If you don't want to add a NULL check, perhaps add a comment at the top of the function stating why.

File: non-GPL/HostIsolationMapsUtil/UpdateMaps.h [hunk 4]
```diff
@@ -6,9 +6,11 @@
  * you may not use this file except in compliance with the Elastic License 2.0.
  */
 
+#ifndef EBPF_UPDATE_MAPS_H
+#define EBPF_UPDATE_MAPS_H
```
Comment (dwhyrock) [inline non-GPL/HostIsolationMapsUtil/UpdateMaps.h:10]: Using `#pragma once` could simplify this.

**Review by dwhyrock — APPROVED** [review]
A couple comments, but nothing to hold up this PR.

**Comment by stanek-michal** [comment]
> A couple comments, but nothing to hold up this PR.

I'll make a note to fix those in the next PR if you don't mind.

**Comment by fntlnz** [comment]
Good job @stanek-michal  I did a review to this and code looked good to me.
 I didn’t have a chance (had issues with my environment)  to run the prog test run tests in this repo and see if they cover the new use case.

I’ll test those and in case there are any tests not passing I’ll open an issue. I suspect not however.

**Pull request merged**